### PR TITLE
docs: add v3.3.0 changelog entries

### PR DESCRIPTION
**Pull request by MSalopek**
## Description

Closes: #XXXX

Add changelog file and cleanup entries.

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [x] included the correct `docs:` prefix in the PR title
- [x] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/interchain-security/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [x] reviewed "Files changed" and left comments if necessary 
- [x] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] Confirmed the correct `docs:` prefix in the PR title
- [ ] Confirmed all author checklist items have been addressed 
- [ ] Confirmed that this PR only changes documentation
- [ ] Reviewed content for consistency
- [ ] Reviewed content for spelling and grammar
- [ ] Tested instructions (if applicable)
- [ ] Checked that the documentation website can be built and deployed successfully (run `make build-docs`)

### CODE REVIEW

**Comment by MSalopek** [comment]
Linter failing is fine, it complains about `uintPtr` helper function from e2e tests - this is an artifact left over from removing provider throttling  v2 changes.

File: .changelog/v3.3.0/improvements/1324-consumer-genesis.md
[the remark this answers is not in the document]
Comment (MSalopek) [reply .changelog/v3.3.0/improvements/1324-consumer-genesis.md:1]: Top of the page: https://github.com/cosmos/interchain-security/commits/release/v3.3.x?after=a5f0123b37e58f204342f6a75f2381a2a6cab04a+34&branch=release%2Fv3.3.x&qualified_name=refs%2Fheads%2Frelease%2Fv3.3.x

File: .changelog/v3.3.0/summary.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+*December 4th, 2023*
```
[the remark this answers is not in the document]
Comment (MSalopek) [reply .changelog/v3.3.0/summary.md:1]: I added the release highlights, I can remove them but they do no harm.

**Review by mpoke — APPROVED** [review]

File: .changelog/v3.3.0/summary.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (mpoke) [reply .changelog/v3.3.0/summary.md:1]: I just want to keep things simple and have release highlights in the release notes, while the changelog contains a list of relevant changes. Not that important though

**Comment by MSalopek** [comment]
Closing in favor of https://github.com/cosmos/interchain-security/pull/1476

**Pull request closed without merging**